added angular 1.x sample

basarat opened the pull request:
As requested in https://github.com/Microsoft/TypeScriptSamples/issues/12

Details available in README.md : https://github.com/basarat/TypeScriptSamples/blob/angular1/angular1/README.md which is a part of this PR :rose:

mhegazy commented on angular1/README.md:
````diff
@@ -0,0 +1,45 @@
+```
+and visit : http://localhost:8000/app/
+
+## Upgrading
````
is this needed for the readme? on another note "upgrading" sounded to vague to me.

mhegazy commented on angular1/README.md:
````diff
@@ -0,0 +1,45 @@
+Start the TypeScript compiler in watch mode (either in the `app` folder or in the `e2e-tests` folder) and **leave it running**: 
+
+```bash
+tsc --watch --p . 
````
i do not see a tsconfig at the root.., you mean `--p app\`

mhegazy commented on angular1/package.json:
```diff
@@ -0,0 +1,38 @@
+  "name": "angular-seed",
+  "private": true,
+  "version": "0.0.0",
+  "description": "A starter project for AngularJS",
```
nit: A starter project for AngularJS using TypeScript

mhegazy commented on angular1/package.json:
```diff
@@ -0,0 +1,38 @@
+  "private": true,
+  "version": "0.0.0",
+  "description": "A starter project for AngularJS",
+  "repository": "https://github.com/angular/angular-seed",
```
https://github.com/Microsoft/TypeScriptSamples/angular1

mhegazy commented on angular1/app/components/version/interpolate-filter.js:
```diff
@@ -0,0 +1,7 @@
+'use strict';
```
I do not think we need the .js files in the sample, it would be confusing i would say.. and they should be generated anyways. and the other samples do not have them

yuit replied:
Agree

mhegazy commented on angular1/app/tsconfig.json:
```diff
@@ -0,0 +1,29 @@
+    "compilerOptions": {
+        "module": "commonjs"
+    },
+    "files": [
```
i do not think you need these. i would just remove them.

mhegazy commented on angular1/app/view1/view1.ts:
```diff
@@ -0,0 +1,19 @@
+'use strict';
+
+class View1Ctrl{
```
View1Controller

mhegazy commented on angular1/bower.json:
```diff
@@ -0,0 +1,15 @@
+  "name": "angular-seed",
+  "description": "A starter project for AngularJS",
+  "version": "0.0.0",
+  "homepage": "https://github.com/angular/angular-seed",
```
same as package.json

mhegazy commented:
Thanks @basarat :sunflower:; just a few comments.

basarat commented:
Addressed the comments: 

* [x] Renamed Upgrading to Conversion as that is what its really documenting. Moved to a separate document "CONVERSION.md".
* [x]  Updated readme to allow the user to run the commands from `root` instead of needed to `cd` into `app` / `e2e-tests` 
* [x] Updated Description and homepage in package.json / bower.json
* [x] Ignored (and removed) the JS files
* [x] Removed redundant contents (`files`) from the `app`'s `tsconfig.json` file. However the `files` listing is needed for the e2e tests as it links back to the files in the `app` directory which is *outside* the directory of the e2e-tests directory. 
* [x] `View1Ctrl` => `View1Controller` (even though the name came from the official angular-seed project : https://github.com/angular/angular-seed/blob/master/app/view1/view1.js#L12-L14)

yuit commented on angular1/README.md:
````diff
@@ -0,0 +1,28 @@
+tsc --watch --p e2e-tests
+```
+
+That's it. You have typescript setup and ready to go. You can follow the steps of JavaScript ([README-JS](./README-JS.md)) from this point on in a new window starting at the install dependencies section.
````
Nit: You can, then, follow the standard steps for seeded AngularJS web app in a new window, using [README-JS](./README-JS.md) starting at the section, [Install Dependencies](https://github.com/basarat/TypeScriptSamples/blob/angular1/angular1/README-JS.md#install-dependencies).

yuit commented on angular1/README.md:
```diff
@@ -0,0 +1,28 @@
+
+That's it. You have typescript setup and ready to go. You can follow the steps of JavaScript ([README-JS](./README-JS.md)) from this point on in a new window starting at the install dependencies section.
+
+**TIP**: *Abbriged the remaining JS steps for a quick start:*
```
Nit: Abridged

yuit commented:
Thanks @basarat. The instruction is very easy to follow when I tried it out :) Just small suggestions otherwise the PR look good :+1: :tulip:

RyanCavanaugh commented on angular1/app/components/version/interpolate-filter.ts:
```diff
@@ -0,0 +1,9 @@
+
+angular.module('myApp.version.interpolate-filter', [])
+
+.filter('interpolate', ['version', function(version) {
```
I think it'd be great if this sample worked under `--noImplicitAny`. Maybe in a future PR

mhegazy replied:
+1

mhegazy commented:
Thanks again @basarat !
